backend/src/server_process_helper.py

```diff
@@ -47,21 +47,28 @@ def __init__(self, flags: list[str]):
         self._stop_event = threading.Event()
 
         # Create a separate thread to read and print the output of the subprocess
-        threading.Thread(
+        self.t1 = threading.Thread(
```

Comment:
`t1` is a terrible name. Since you already named the thread "output reader", maybe we could call the variable `_reader_thread`.